### ISSUE TITLE
block gzipped VCF not recognized as VCF

### ISSUE THREAD

**Issue by akiezun** [HadoopGenomics/Hadoop-BAM#68]
I have a block gzipped vcf and I want to read it using a call like this:
```
ctx.newAPIHadoopFile(
                vcf, VCFInputFormat.class, LongWritable.class, VariantContextWritable.class,
                new Configuration());
```

`VCFFormat.inferFromData` falsely thinks it's a BCF because it's gzipped and starts with a `1f` byte. And so my code breaks. The vcf is attached. Block gzipped VCF should be recognized as VCFs not BCFs. 

[count_variants.blockgz.gz](https://files.zenhub.io/56c4e45d5d3a5c4e4bb3ccf5)
Uploaded using [**ZenHub.io**](http://www.zenhub.io)

**Comment by tomwhite** [comment]
@akiezun I created a fix in #70. Could you test to see if it works for you?

**Comment by akiezun** [comment]
thanks @tomwhite - does it work for blocked gzipped files? (we typically use bgzip from samtools). http://www.htslib.org/doc/tabix.html 
You can use the file I attached: count_variants.blockgz.gz

**Comment by tomwhite** [comment]
I posted some new code in #70. I tested it with count_variants.blockgz.gz and it could read the file fine. The PR has more tests, including a large file with multiple splits.

**Comment by tomwhite** [comment]
Fixed in #70.
